File: backend/app/types/item.go
package types

type APIItem struct {
	ID           uint64 `json:"id"`
	Title        string `json:"title"`
	Image        string `json:"image"`
	Category     string `json:"category"`
	CategoryName string `json:"category_name"`
	Price        uint64 `json:"price"`
	Creator      uint64 `json:"creator"`
	Seller       bool   `json:"seller"`
	Purchased    bool   `json:"purchased"`
	CartAdded    bool   `json:"cart_added"`
}

type APIItems struct {
	Quantity int       `json:"quantity"`
	Items    []APIItem `json:"items"`
}

File: frontend/src/types/item.ts
export type ItemType = {
  id: number
  title: string
  image: string
  category: string
  category_name: string
  price: number
  creator: number
  seller: boolean
  purchased: boolean
  cart_added: boolean
}

export type ItemsType = {
  quantity: number
  items: Array<ItemType>
}
